src/stm32f1/gpio.c

```diff
@@ -219,3 +221,101 @@ gpio_adc_cancel_sample(struct gpio_adc g)
         LL_ADC_ClearFlag_EOS(ADC1);
     }
 }
+
+/****************************************************************
+ * Serial Peripheral Interface (SPI) pins
+ ****************************************************************/
+
+void spi_set_mode(SPI_TypeDef *spi, uint8_t mode)
+{
+    switch (mode) {
+    case 0:
+        LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_LOW);
+        LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_1EDGE);
+        break;
+    case 1:
+        LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_LOW);
+        LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_2EDGE);
+        break;
+    case 2:
+        LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_HIGH);
+        LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_1EDGE);
+        break;
+    case 3:
+        LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_HIGH);
+        LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_2EDGE);
+        break;
+    default:
+        shutdown("Invalid SPI mode");
+    }
+}
+
+void spi_set_baudrate(SPI_TypeDef *spi, uint32_t rate)
+{
+    const uint32_t pclk = __LL_RCC_CALC_PCLK1_FREQ(SystemCoreClock, LL_RCC_GetAPB1Prescaler());
+    const uint32_t prescaler = pclk / rate;
+
+    if (prescaler > 256 || prescaler < 2)
+        shutdown("Invalid SPI baudrate");
+
+    uint32_t setting = LL_SPI_BAUDRATEPRESCALER_DIV256;
+    if (prescaler <= 128)
+        setting = LL_SPI_BAUDRATEPRESCALER_DIV128;
+    else if (prescaler <= 64)
```

Comment:
Typo?  (Unless I'm missing something, prescaler can't be less than 64 if it is already tested to be less than 128.)